docs/content/docs/internals/job_scheduling.md

```diff
@@ -95,3 +95,22 @@ For that reason, the execution of an ExecutionVertex is tracked in an {{< gh_lin
 {{< img src="/fig/state_machine.svg" alt="States and Transitions of Task Executions" width="50%" >}}
 
 {{< top >}}
+
+## Repeatable Resource Cleanup Strategy
+
+Once a job has reached a terminally global state of either finished, failed or cancelled, the
+resources associated with the job are then cleaned up. This is done via a repeatable
+resource cleanup strategy, in which failures to clean up a resource result retries separated by
+an exponentially increasing delay, within configured bounds.
+
+{{< img src="/fig/repeatable_cleanup.png" alt="Repeatable resource cleanup across a failover event" >}}
```

Comment:
I like that you integrated the picture. :-) I'm just wondering whether that's too detailed for the documentation. We're not talking about the individual components here (or in `Deployments / Overview`). Hence, I was wondering whether we should simplify it and hide the details on what components are handled internally. That would also have the benefit if we're changing something in terms of what components are subject to cleanup. In that case, we wouldn't need to change this picture. 🤔

Reply:
Yeah, in this case perhaps something that abstracted away the actual components would make sense. Thanks for the idea!